File: PRACTICA 1/punto.py
class Punto:
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y

    def coord_cartesianas(self):
        return (self.x, self.y)

    def coord_polares(self):
        import math
        r = math.sqrt(self.x**2 + self.y**2)
        theta = math.atan2(self.y, self.x)
        return (r, theta)

    def __str__(self):
        return f"Punto({self.x}, {self.y})"

if __name__ == "__main__":
    p = Punto(3, 4)
    print("Coordenadas Cartesianas:", p.coord_cartesianas())
    print("Coordenadas Polares:", p.coord_polares())
    print("Representación String:", p)

File: PRACTICA 1/Punto.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompany.punto;

/**
 *
 * @author Lenovo
 */
public class Punto {
    public static void main(String[] args) {
        class Main {
            private double x;
            private double y;

            public Main(double x, double y) {
                this.x = x;
                this.y = y;
            }

            public double[] coordCartesianas() {
                return new double[] {x, y};
            }

            public double[] coordPolares() {
                double r = Math.sqrt(x * x + y * y);
                double theta = Math.atan2(y, x);
                return new double[] {r, theta};
            }

            @Override
            public String toString() {
                return "Punto(" + x + ", " + y + ")";
            }
        }

        Main p = new Main(3, 4);
        double[] cartesianas = p.coordCartesianas();
        System.out.println("Coordenadas Cartesianas: (" + cartesianas[0] + ", " + cartesianas[1] + ")");
        
        double[] polares = p.coordPolares();
        System.out.println("Coordenadas Polares: (r: " + polares[0] + ", θ: " + polares[1] + ")");
        
        System.out.println("Representación String: " + p);
    }
}
